AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -394,13 +391,13 @@ - (void)configAuthenticationClient {
 
   // Init MSAL client application.
   NSError *error;
-  MSALB2CAuthority *auth = [[MSALB2CAuthority alloc] initWithURL:(NSURL * __nonnull) self.authConfig.authorities[0].authorityUrl error:nil];
+  MSALAuthority *auth = [MSALAuthority authorityWithURL:(NSURL * __nonnull)self.authConfig.authorities[0].authorityUrl error:nil];
   MSALPublicClientApplicationConfig *config =
-      [[MSALPublicClientApplicationConfig alloc] initWithClientId:(NSString * __nonnull) self.authConfig.clientId
+      [[MSALPublicClientApplicationConfig alloc] initWithClientId:(NSString * __nonnull)self.authConfig.clientId
                                                       redirectUri:self.authConfig.redirectUri
                                                         authority:auth];
   if (!auth) {
-    MSLogError([MSAuth logTag], @"Auth config doesn't contain a valid default B2C authority.");
+    MSLogError([MSAuth logTag], @"Auth config doesn't contain a valid default %@ authority.", self.authConfig.authorities[0].type);
```

Comment:
As far as I know some authorities might not contain type. Are you sure the first authority always contains type?
Isn't there a chance that `self.authConfig.authorities` could be `nil` or empty?

Reply:
In MSAuthConfig we make sure that the default authority is processed and it contains a valid type either B2C or AAD or we throw error. This preprocessing steps make sure we always have a valid type.

Reply:
First authority here will always be an authority whose default is true and has type = AAD or B2C. MSAuthConfig makes sure of it. If you look at a couple of lines above this code, we send 1st authority to MSAL for authentication